src/utils/react/interactions.ts

```diff
@@ -4,6 +4,7 @@ import {
   useFocus as reactAriaUseFocus,
   useFocusVisible,
 } from '@react-aria/interactions';
+import type {} from "@react-types/shared";
```

Comment:
Why here are we doing empty import?

Reply:
This PR is WIP, I've listed the pending list in the description. I might leave something here as I pushed in b/w the work.

Reply:
Ok! Can't wait to see its final state 👀